motionplan/constraint.go

```diff
@@ -587,11 +566,13 @@ func NewEmptyConstraints() *Constraints {
 // NewConstraints initializes a Constraints object with user-defined LinearConstraint, OrientationConstraint, and CollisionSpecification.
 func NewConstraints(
 	linConstraints []LinearConstraint,
+	pseudoConstraints []PseudolinearConstraint,
```

Comment:
Not a fan of adding this here.  This should be just for the constraints we expose through proto.  You can always append to the Constraints with custom constraints after the fact

Reply:
Actually, not sure on this.

Since this is the structure we use for our own consumption and passing around of constraints and initialization of actual constraint functions, these structures should I think be the union of what we support in proto, and what we support via things like `extra`. This allows us to better prototype what will eventually be added into proto. The alternative is creating a parallel data structure that we pass around to do the same thing, but which adds parameters, masks the putative new API, and makes it easier to forget things.

Reply:
Just saw this second reply now.  I'm forgetting why we have the constraints typed this way in the first place. IIRC it used to just be a slice of Constraints.  My comment makes a lot less sense now that I see this is no longer the case